README.routemaster_client.md

```diff
@@ -8,66 +8,61 @@ It also assumes that your app has an API for the resources you want to publish l
 
 ### Setup lifecycle events for your models
 
-We will most likely want to publish lifecycle events for several models, so to write slightly less code let's create a model concern first:
-
-```ruby
-# app/models/concerns/routemaster_lifecycle_events.rb
-require 'roo_on_rails/routemaster/lifecycle_events'
-
-module RoutemasterLifecycleEvents
-  extend ActiveSupport::Concern
-  include RooOnRails::Routemaster::LifecycleEvents
-
-  included do
-    publish_lifecycle_events
-  end
-end
-```
-
-Then let's include this concern to the relevant model(s):
+You can use publish events on create, update, and destroy by including the `PublishLifecycleEvents` module:
 
 ```ruby
 # app/models/order.rb
+require 'roo_on_rails/routemaster/publish_lifecycle_events'
+
 class Order < ApplicationRecord
-  include RoutemasterLifecycleEvents
+  include RooOnRails::Routemaster::PublishLifecycleEvents
 
   # ...
 end
 ```
 
-And another one for the example:
+If you need more control over which events are published you can use the base module `LifecycleEvents` and specify them explicitly:
 
 ```ruby
 # app/models/rider.rb
+require 'roo_on_rails/routemaster/publish_lifecycle_events'
+
 class Rider < ApplicationRecord
-  include RoutemasterLifecycleEvents
+  include RooOnRails::Routemaster::LifecycleEvents
+
+  publish_lifecycle_events :create, :destroy
 
   # ...
 end
 ```
 
 ### Create publishers for lifecycle events
 
-We have now configured our models to publish lifecycle events to Routemaster, but it won't send anything until you have enabled publishing and created matching publishers. Let's start with creating a `BasePublisher` that we can then inherit from:
+We have now configured our models to publish lifecycle events to Routemaster, but it won't send anything until you have enabled publishing and created matching publishers. Let's start with creating an `ApplicationPublisher` that we can use as our default.
 
 ```ruby
-# app/publishers/base_publisher.rb
+# app/publishers/application_publisher.rb
 require 'roo_on_rails/routemaster/publisher'
 
-class BasePublisher < RooOnRails::Routemaster::Publisher
+class ApplicationPublisher < RooOnRails::Routemaster::Publisher
   include Rails.application.routes.url_helpers
 
+  def url
+    url_helper = :"api_#{model.class.name.underscore}_url"
+    public_send(url_helper, model.id, host: ENV.fetch('API_HOST'), protocol: 'https')
```

Comment:
Now I'm thinking that the documentation's example should probably use UUIDs as it's what we've decided to go with and as it requires a [little trick](https://github.com/deliveroo/logistics-dashboard/pull/583/files#diff-38264488dcf6826516388149f59bbcbc) to work correctly.